modules/planning/lattice/trajectory_generation/trajectory1d_generator.cc

```diff
@@ -39,11 +39,12 @@ namespace {
 
 // A common function for trajectory bundles generation with 
 // a given initial and  end conditions.
-typedef std::array<double, 3> Init_Condition_T;
-typedef std::vector<std::pair<std::array<double, 3>, double>> End_Conditions_T;
+typedef std::array<double, 3> InitCondition;
+typedef std::vector<std::pair<std::array<double, 3>, double>> EndConditions;
+template <class PolynomialCurve1dType>
```

Comment:
This template does not solve the problem yet, because the details of dealing with quintic and quadratic are different. Please double check the details. I recommend you can implement a common function for quadratic polynomial only, because quadratic polynomial is majority. Thanks for your effort.

Reply:
Sorry, type, quadratic -> quartic